app/src/main/java/com/example/plantsapp/data/usecase/AuthUseCaseImpl.kt

```diff
@@ -10,16 +12,20 @@ import java.lang.IllegalStateException
 import javax.inject.Inject
 
 class AuthUseCaseImpl @Inject constructor(
-    private val auth: FirebaseAuth
+    private val auth: FirebaseAuth,
+    @FirebaseQualifier private val userRepository: UserRepository
 ) : AuthUseCase {
 
     @Throws(IllegalStateException::class)
-    override suspend fun invoke(input: AuthUseCase.AuthInput): User {
-        return getSignedInFirebaseUser(input.token)
+    override suspend fun invoke(input: AuthUseCase.AuthInput) {
+        getSignedInFirebaseUser(input.token)
             ?.let {
-                User(
-                    name = it.displayName!!,
-                    profilePicture = it.photoUrl
+                userRepository.setUser(
+                    User(
+                        uid = it.uid,
+                        name = it.displayName ?: "No Name",
```

Comment:
hmm, still not ok to leave hardcoded strings here

Im not in context about this displayName, but how about
1. Remove it completely for now
2. leave !! but leave TODO comment
3. Or throw some more concrete exception (if it ok to do so)